moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -488,7 +488,7 @@ class PlanningSceneMonitor : private boost::noncopyable
   std::unique_ptr<boost::thread> publish_planning_scene_;
   double publish_planning_scene_frequency_;
   SceneUpdateType publish_update_types_;
-  SceneUpdateType new_scene_update_;
+  std::atomic<SceneUpdateType> new_scene_update_;
```

Comment:
I'm wondering if this needs to be fixed for ROS1 as well... or if we have some bugs with the mutex locks in here

Reply:
The issue here is that on one thread we treat this as part of a condition variable and are checking it's state in a loop.  In another thread we do a RMW (read-modify-write) operation on it.  By telling the compiler that this is atomic we make the RMW operation atomoic.

Reply:
That part is exactly the same as with MoveIt 1, we should probably backport this